pymatgen/core/interface.py

```diff
@@ -392,7 +397,7 @@ def gb_from_parameters(
         ab_shift: tuple[float, float] = (0, 0),
         normal: bool = False,
         ratio: list[int] | None = None,
-        plane: list[int] | None = None,
+        plane: tuple[int, int, int] | None = None,
```

Comment:
Similar to the kpoints type, I have to change `plane` to `tuple[int, int, int]` in order to track its length. I'm hoping no code would reply on its type.

Reply:
given it's always length 3, a tuple would have been the better type here from the start

Reply:
Thanks for confirming, and the same applies to several other variables:
```python
    @staticmethod
    def get_trans_mat(
        r_axis: tuple[int, int, int] | tuple[int, int, int, int],
        angle: float,
        normal: bool = False,
        trans_cry: NDArray | None = None,
        lat_type: str = "c",
        ratio: list[int] | None = None,
        surface: tuple[int, int, int] | tuple[int, int, int, int] | None = None,
        max_search: int = 20,
        quick_gen: bool = False,
    ):
```

I'm wondering however, is there a more efficient way to express the length of the `tuple` when it could be one of several possible lengths, other than the super long `r_axis: tuple[int, int, int] | tuple[int, int, int, int],`?

Reply:
you could define type variables:

```py
Tuple3Ints = tuple[int, int, int]
Tuple4Ints = tuple[int, int, int, int]
Tuple3Ints | Tuple4Ints
```

Reply:
Yes this is a more concise way :) typing `tuple[int, ...]` is making the line length break the roof.
Would implement this in this PR

Reply:
Yep, we're thinking the same :)